src/encounter.c

```diff
@@ -40,116 +42,128 @@
-        }
-        break;
+        case 0:
+            sub_0203E3C4(fieldSystem, battleSetup);
+            sub_0203E354();
+            (*state)++;
+            break;
+        case 1:
+            if (!FieldSystem_ApplicationIsRunning(fieldSystem)) {
+                return TRUE;
+            }
+            break;
     }
     return FALSE;
 }
 
-static void sub_020506AC(TaskManager *man, BATTLE_SETUP *setup) {
-    TaskManager_Call(man, sub_02050660, setup);
+static void sub_020506AC(TaskManager *taskManager, BattleSetup *setup) { //CallTask_StartBattle
+    TaskManager_Call(taskManager, sub_02050660, setup);
 }
 
-static ENCOUNTER *Encounter_New(BATTLE_SETUP *setup, int effect, int bgm, u32 *flag) {
-    ENCOUNTER *work;
-    work = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(ENCOUNTER));
-    work->winFlag = flag;
-    if (flag != NULL) {
-        *flag = 0;
+static Encounter *Encounter_New(BattleSetup *setup, s32 effect, s32 bgm, u32 *winFlag) {
+    Encounter *encounter = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(Encounter));
+    encounter->winFlag = winFlag;
+    if (winFlag != NULL) {
+        *winFlag = 0;
     }
-    work->effect = effect;
-    work->bgm = bgm;
-    work->setup = setup;
-    return work;
+    encounter->effect = effect;
+    encounter->bgm = bgm;
+    encounter->setup = setup;
+    return encounter;
 }
 
-static void Encounter_Delete(ENCOUNTER *work) {
-    BattleSetup_Delete(work->setup);
-    FreeToHeap(work);
+static void Encounter_Delete(Encounter *encounter) {
+    BattleSetup_Delete(encounter->setup);
+    FreeToHeap(encounter);
 }
 
-static BOOL sub_020506F4(ENCOUNTER *work, FieldSystem *fsys) {
-    if (work->winFlag != NULL) {
-        *(work->winFlag) = work->setup->winFlag;
+static BOOL GetEncounterResult(Encounter *encounter, FieldSystem *fieldSystem) {
+    if (encounter->winFlag != NULL) {
+        *(encounter->winFlag) = encounter->setup->winFlag;
     }
-    VarSet(fsys, VAR_BATTLE_RESULT, work->setup->winFlag);
-    return IsBattleResultWin(work->setup->winFlag);
+    VarSet(fieldSystem, VAR_BATTLE_RESULT, encounter->setup->winFlag);
+    return IsBattleResultWin(encounter->setup->winFlag);
 }
 
-static void sub_02050724(BATTLE_SETUP *setup, FieldSystem *fsys) {
+static void sub_02050724(BattleSetup *setup, FieldSystem *fieldSystem) {
     if (!(setup->flags & BATTLE_TYPE_DEBUG)) {
-        sub_0205239C(setup, fsys);
+        sub_0205239C(setup, fieldSystem);
     }
 }
 
-static BOOL sub_02050738(TaskManager *man) {
-    FieldSystem *fsys = TaskManager_GetFieldSystem(man);
-    ENCOUNTER *encounter = TaskManager_GetEnv(man);
-    int *state = TaskManager_GetStatePtr(man);
+static BOOL Task_StartEncounter(TaskManager *taskManager) { //todo: better name
+    FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskManager);
+    Encounter *encounter = TaskManager_GetEnvironment(taskManager);
+    u32 *state = TaskManager_GetStatePtr(taskManager);
 
     switch (*state) {
         case 0:
```

Comment:
Style: State machines should have constants defined for them to make them easier to modify

Reply:
etc for all state machines in this file